### ISSUE TITLE
[discuss]Should we remove the "+" icon for single metric control?

### ISSUE THREAD

**Issue by ktmud** [apache/superset#12631]
## Screenshot

<img src="https://user-images.githubusercontent.com/335541/105228368-5eee8f00-5b17-11eb-81f7-7d45f7d5ba30.png" width="320">

<img src="https://user-images.githubusercontent.com/335541/105228392-64e47000-5b17-11eb-9557-026638d9eb29.png" width="320">

## Description

Should we remove "+" icon for the the single metric control? It's not adding much value and kind of distracting. 

Maybe we can even get rid of the "+" altogether by always keeping a "Add metric"/"Add filter" CTA in the selected area:

<img src="https://user-images.githubusercontent.com/335541/105228937-108dc000-5b18-11eb-91e5-0f786fb9c9b0.png" width="320">

We already received user feedbacks that they did not know how to add a second metric.

## Design input

Just a proposal

**Comment by junlincc** [comment]
Thanks for proposing! Could you provide some actual quotes from users about their confusion? I would also like to meet with them if its possible. When it comes to UI/UX we wanna understand the root problem first before diving into any solutions. Also please keep in mind that we are in transition phase, getting ready to fully implement the proposed design in SIP 34. Nothing is not done and set, we welcome feedbacks from all organization.

**Comment by ktmud** [comment]
"I can’t figure out how to add a second metric" is the exact quote.

**Comment by srinify** [comment]
@junlincc I will say, in general it took me many attempts to understand the + button. I found it very awkward that it was floating above / in the top right corner of the metric name. I actually expected something like the solution proposed here (a persistent ghost-option / shadow button to add a new metric)

**Comment by junlincc** [comment]
reasons why I have some hesitation 
- Original design is proposed by Cartel in SIP 34 with a fair amount of research went it. it has served as a design Northstar in the product
- UI change can be subjective; we haven't heard enough 'complaint' yet on this subject. Let's let it sit for a while. When we start seeing a pattern of preference, we can get back to it
- Superset design system guidelines are not fully baked yet.  Until they are ready, we try to avoid changing back and forth, which could lead to  diminishing return, especially in the area(Query control) where it is mostly visited/used. 

By knowing both of you, who have close contact with large number of users and knowledge in BI tool, having the same feedback, I'm pretty convinced already.  @ktmud @srinify 

cc @mihir174 , bringing it to your attention; no immediate action needed yet

**Comment by villebro** [comment]
I personally think this proposal is an improvement for the following reasons:
- the surface area of the "Add" event is larger and requires less aiming (the current plus button is pretty tiny)
- the new metric shows up exactly where the user clicked in the first place (less confusion)
- the user can click on the ghost button as many times as they want until they run out of ghost buttons (=viz doesn't support more metrics) - no need to inspect the color of the plus button.
- The ghost button just feels very intuitive, almost calling on the user to press it 😆 

So voting +1 for this.

**Comment by junlincc** [comment]
🤣 🤣 i vote +1 too, just delay implementing the change for a bit....we are lacking design support severely at this moment and i don't feel confident/comfortable enough to make ad hoc changes this way. please understand guys

**Comment by villebro** [comment]
I agree with @junlincc , let's let this go through proper design review so we don't cause unnecessary back and forth.

**Comment by zhaoyongjie** [comment]
> I agree with @junlincc , let's let this go through proper design review so we don't cause unnecessary back and forth.

I am totally in agreement with that.

**Comment by ktmud** [comment]
Hi, @srinify and @villebro , thanks for the inputs! Most of my UX suggestions are based on my intuition, and sometimes I may have not explained myself too well. It's nice to see someone else articulate the benefits of a proposed design better than me.

I agree we don't have to implemented this right away and I never expected it to be. The whole thing is just a proposal and if 
there are strong reasons of objections or better alternatives, I'd be happy to drop it.

But I don't believe this particular proposal is blocked by design resources as the proposal is pretty clear/simple and basically just an extension of the existing design. As an open source project, a lot of discussions for Superset happens asynchronously like this on Github issues. "Proper design review" would be a luxury for most cases. I'd be happy to hear designers' opinion on this, but sometimes we should not have to wait for it, exactly because how severely constrained the design resource is.

Also comment on these two points:

> 1. Original design is proposed by Cartel in SIP 34 with a fair amount of research went it. it has served as a design Northstar in the product

IMO, SIP-34 is more like a north star for design atheistic, rather than every detail of all the UX. Nothing is set on stone. It's quite natural that issues come up when a design mockup is being actually implemented. We should be more flexible here and open to changes. Plus, I could be wrong, but SIP-34 mockup doesn't even have an empty state for these controls, indicating this area was probably not carefully considered to begin with.

> Superset design system guidelines are not fully baked yet. Until they are ready, we try to avoid changing back and forth

IMO this is a step forward, not back. Unless we believe the "+" icon will have to come back at some point, it's relatively low-risk. It's true introducing new UX patterns may break user habits they just got used to, but his particular proposal is not a new pattern as users would be quite familiar with the ghost button anyway.

**Comment by junlincc** [comment]
you have made your points loud and clear, I agree with most of them. 
I believe Explore control is the most frequently used feature in the entire product, therefore it deserves a full cycle of research-design-implementation-testing, which takes time. All I'm asking here is to allow that process to happen. among all the other design tasks, I consider this one is low priority since it's not blocking any operation, while others could. also we should leave some time for other users to provide feedback since we just roll out v1.0. We could probably come up with even better solution from the collective wisdom than simply removing the "+".

**Comment by ktmud** [comment]
@junlincc Agree with most of what you said and totally understand the desire of having a stable product and making decisions based on more user feedbacks.

However, this design does block some new users from adding a second metric, at least initially, as evidenced by our user feedbacks. I'd consider anything that already has a negatively user feedback (even just one) to be relatively high priority... We don't have to agree with every user feedback, but we should at least consider if there are better solutions to their problems. If yes, then why not go ahead with it? If someone comes up with a better solution later, we can just iterate.

I guess what I didn't fully get is why can't we put this into the "OK to implement, but no need to prioritize" bucket, rather than "hold until further design review which nobody know when will happen" bucket, when there is already enough consensus?

**Comment by mihir174** [comment]
Hey everyone, this was a very interesting discussion. Here's my take:

A big "plus" side to the button is that the control panel has a large vertical scroll surface, so having an "Add metric" CTA in the selected area will only make this taller and require more user scrolling. That being said, I agree that the button is unnecessary for input fields that take only 1 input.

Given that we have received feedback from a couple of users, I think it's worth running some basic usability testing to compare different options before coming to a conclusion. If we are confident that users will prefer the proposed design change and if we think this issue is critical enough to address immediately, I will also +1 the change (bearing in mind the added vertical real estate).

**Comment by rusackas** [comment]
The [+] buttons appear to be on a path toward deprecation in Superset 3.0, as the two Feature Flags for drag and drop controls are now on by default, and there have been few to no support requests for the older code paths. For that reason, I'll go ahead and close this issue, but if you think this is overstepping, I'm happy to rekindle the conversation here, or in a Discussion thread, whatever works. Thanks, as always!
